sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -477,5 +486,82 @@ object StateStore extends Logging {
       None
     }
   }
-}
 
+  private class StateSchemaCompatibilityChecker(
```

Comment:
I like that it's a separate class because it has different responsibility, maybe it can be a separate file?